src/range-slider/index.ts

```diff
@@ -0,0 +1,372 @@
+		minEvent && minEvent.stopPropagation();
+		maxEvent && maxEvent.stopPropagation();
+
+		const { min = 0, max = 100 } = this.properties;
+		const { minValue = min, maxValue = max } = this.properties;
+
+		callback && callback(
+			minEvent ? extractValue(minEvent) : minValue,
+			maxEvent ? extractValue(maxEvent) : maxValue
+		);
+	}
+
+	private _genericChangeCallback(callback?: MinMaxCallback, minEvent?: Event, maxEvent?: Event) {
+		minEvent && minEvent.stopPropagation();
+		maxEvent && maxEvent.stopPropagation();
+
+		const { min = 0, max = 100 } = this.properties;
+		const { minValue = min, maxValue = max } = this.properties;
+
+		if (minEvent) {
+			callback && callback(Math.min(extractValue(minEvent), maxValue), maxValue);
+		}
+		else if (maxEvent) {
+			callback && callback(minValue, Math.max(minValue, extractValue(maxEvent)));
+		}
+	}
+
+	private _onKeyDown(event: KeyboardEvent) {
+		event.stopPropagation();
+		this.properties.onKeyDown && this.properties.onKeyDown(event.which, () => { event.preventDefault(); });
+	}
+
+	private _onKeyPress(event: KeyboardEvent) {
+		event.stopPropagation();
+		this.properties.onKeyPress && this.properties.onKeyPress(event.which, () => { event.preventDefault(); });
+	}
+
+	private _onKeyUp(event: KeyboardEvent) {
+		event.stopPropagation();
+		this.properties.onKeyUp && this.properties.onKeyUp(event.which, () => { event.preventDefault(); });
+	}
+
+	private _onMouseDown(event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onMouseDown && this.properties.onMouseDown();
+	}
+
+	private _onMouseUp(event: MouseEvent) {
+		event.stopPropagation();
+		this.properties.onMouseUp && this.properties.onMouseUp();
+	}
+
+	private _onTouchStart(event: TouchEvent) {
+		event.stopPropagation();
+		this.properties.onTouchStart && this.properties.onTouchStart();
+	}
+
+	private _onTouchEnd(event: TouchEvent) {
+		event.stopPropagation();
+		this.properties.onTouchEnd && this.properties.onTouchEnd();
+	}
+
+	private _onTouchCancel(event: TouchEvent) {
+		event.stopPropagation();
+		this.properties.onTouchCancel && this.properties.onTouchCancel();
+	}
+
+	private _getInputProperties(isSlider1: boolean) {
+		const {
+			aria = {},
+			disabled,
+			invalid,
+			max = 100,
+			min = 0,
+			name = '',
+			readOnly,
+			required,
+			step = 1,
+			minimumLabel = 'Minimum',
+			maximumLabel = 'Maximum'
+		} = this.properties;
+		const {
+			minName = `${name}_min`,
+			maxName = `${name}_max`
+		} = this.properties;
+
+		const prepareCallback = (callback: (callback?: MinMaxCallback, minEvent?: Event, maxEvent?: Event) => void, property?: MinMaxCallback) => {
+			return (e?: Event) => {
+				callback(property, ...[isSlider1 ? e : undefined, !isSlider1 ? e : undefined]);
+			};
+		};
+
+		return {
+			...formatAriaProperties(aria),
+			'aria-invalid': invalid === true ? 'true' : null,
+			'aria-readonly': readOnly === true ? 'true' : null,
+			'aria-label': isSlider1 ? minimumLabel : maximumLabel,
```

Comment:
Sorry I didn't catch this before! This is a pretty good solution, but I think the inputs need to be labeled by the `label` property's text in some way. Since it's not simple to get the plain text from `label`, it might be better to point `aria-labelledby` from both of the `input`s to the `label`, then use `aria-describedby` for min/max text? I really wish there were an `aria-description` attribute :/